File: infra/src/main/java/com/beat/infra/InfraBaseConfig.java
package com.beat.infra;

/**
 * Marker for top-level infra bootstrap configurations selectable through
 * {@link InfraBaseConfigGroup}.
 *
 * <p>Support configurations imported or scanned by those top-level groups must
 * not implement this marker; only enum-owned entrypoint configs should.
 */
public interface InfraBaseConfig {
}

File: infrastructure/src/main/kotlin/com/beat/infrastructure/InfraBaseConfig.kt
package com.beat.infrastructure

/**
 * Marker for top-level infra bootstrap configurations selectable through
 * [InfraBaseConfigGroup].
 *
 * Support configurations imported or scanned by those top-level groups must
 * not implement this marker; only enum-owned entrypoint configs should.
 */
internal interface InfraBaseConfig
